File: 0147-insertion-sort-list/0147-insertion-sort-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* insertionSortList(ListNode* head) {
        int swap_count;
        while(swap_count!=0){
            swap_count = 0;
            ListNode* node = head;
            while(node!=nullptr&&node->next!=nullptr){
                if(node->val > node->next->val) {
                    int temp = node->val;
                    node->val = node->next->val;
                    node->next->val=temp;
                    swap_count++;
                }
                node=node->next;
            }
        }
        return head;
    }
};

File: 0147-insertion-sort-list/0147-insertion-sort-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def insertionSortList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head:
            return head
        
        dummy = ListNode(0)
        current = head
        prev = dummy
        next_ = None
        while current:
            next_ = current.next
            while prev.next and prev.next.val < current.val:
                prev = prev.next
                
            current.next = prev.next
            prev.next = current
            prev = dummy
            current = next_
        return dummy.next
        